File: Java/lab17/src/main/java/org/example/ConsultantAbstract.java
package org.example;

public abstract class ConsultantAbstract {
    protected IMediator mediator;
    protected String name;
    protected String role;

    public ConsultantAbstract(IMediator mediator, String name, String role) {
        this.mediator = mediator;
        this.name = name;
        this.role = role;
    }
    public abstract void sendOrder(Order order);
    public abstract void getOrder(Order order);

}

File: Python/lab17/ConsultantAbstract.py
from IMediator import IMediator
from abc import ABC, abstractmethod
from Order import Order


class ConsultantAbstract(ABC):
    def __init__(self, mediator, name, role):
        self.mediator = mediator
        self.name = name
        self.role = role

    @abstractmethod
    def send_order(self, order: Order):
        pass

    @abstractmethod
    def get_order(self, order: Order):
        pass
